src/js/components/common/enterpriseNotification.js

```diff
@@ -11,23 +11,87 @@
 //    See the License for the specific language governing permissions and
 //    limitations under the License.
-import React from 'react';
-
-import InfoHint from './info-hint';
-
-const EnterpriseNotification = ({ isEnterprise, benefit }) =>
-  !isEnterprise && (
-    <InfoHint
-      content={
-        <span>
-          {`With a more advanced commercial Mender plan you get access to ${benefit}. Get an overview of the Mender plans on the `}
-          <a href="https://mender.io/plans/features" target="_blank" rel="noopener noreferrer">
-            features page
-          </a>
-          .
-        </span>
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
+import { Link } from 'react-router-dom';
+
+import { Chip } from '@mui/material';
+import { withStyles } from 'tss-react/mui';
+
+import { ADDONS, BENEFITS, PLANS } from '../../constants/appConstants';
+import { getTenantCapabilities } from '../../selectors';
+import MenderTooltip, { MenderTooltipClickable } from './mendertooltip';
+
+const PlansTooltip = withStyles(MenderTooltip, ({ palette }) => ({
+  arrow: {
+    color: palette.tooltip.tierTipBackground
+  },
+  tooltip: {
+    backgroundColor: palette.tooltip.tierTipBackground,
+    maxWidth: 300
+  }
+}));
+
+const PlanChip = withStyles(Chip, ({ palette }) => ({
+  root: {
+    backgroundColor: palette.tooltip.tierTipBackground,
+    color: palette.text.disabled,
+    textTransform: 'uppercase',
+    '&:hover': {
+      fontWeight: 'bold'
+    }
+  }
+}));
+
+export const DefaultUpgradeNotification = props => (
+  <div {...props}>
+    This feature is not available on your plan. <Link to="/settings/upgrade">Upgrade</Link> to enable it
+  </div>
+);
+
+const EnterpriseNotification = ({ className = '', id = BENEFITS.default.id }) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const tenantCapabilities = useSelector(getTenantCapabilities);
+  const { isEnterprise, plan: currentPlan } = tenantCapabilities;
+  const { benefit, requiredAddon = '', requiredPlan = PLANS.os.id } = BENEFITS[id];
+  const hasAddon = requiredAddon ? ADDONS[requiredAddon].needs.every(need => tenantCapabilities[need]) : false;
+
+  const currentPlanIndex = Object.keys(PLANS).indexOf(currentPlan);
+  const requiredPlanIndex = Object.keys(PLANS).indexOf(requiredPlan);
+  const shouldShow = requiredPlanIndex > currentPlanIndex;
+  // we have to explicitly check for the plan requirement here, since the default value prevents us from relying on the `shouldShow` result
+  if (isEnterprise || (BENEFITS[id].requiredPlan && !shouldShow) || (requiredAddon && hasAddon)) {
+    return null;
+  }
+  const content = requiredAddon ? (
+    <>
+      Add the <b>{ADDONS[requiredAddon].title}</b> add-on to {benefit}.
+    </>
+  ) : (
+    <>
+      Upgrade to the <b>{PLANS[requiredPlan].name}</b> plan {requiredPlanIndex === Object.keys(PLANS).length - 1 ? '' : 'or higher '}to {benefit}.
```

Comment:
Small adjustment needed to improve the text here: maybe "gain access to {benefit}". 

See screenshot: 
![image](https://github.com/mendersoftware/gui/assets/15045782/7bd22a8f-2e0f-40fd-9904-818820c3b510)